src/components/Timeline/TimelineAddNote.tsx

```diff
@@ -0,0 +1,64 @@
+import { Collapse } from '@material-ui/core';
+import { useIntl } from 'react-intl';
+import React, { useEffect, useState } from 'react';
+
+import SubmitCancelButtons from '../forms/common/SubmitCancelButtons';
+import TextEditor from './TextEditor';
+import { ZetkinNote } from 'types/zetkin';
+
+interface AddNoteProps {
+  disabled?: boolean;
+  onSubmit: (note: Pick<ZetkinNote, 'text'>) => void;
+}
+
+const TimelineAddNote: React.FunctionComponent<AddNoteProps> = ({
+  disabled,
+  onSubmit,
+}) => {
+  const intl = useIntl();
+  const [clear, setClear] = useState<number>(0);
+  const [note, setNote] = useState<Pick<ZetkinNote, 'text'> | null>(null);
+
+  useEffect(() => {
+    if (!disabled) {
+      onCancel();
+    }
+  }, [disabled]);
+
+  return (
+    <form
+      onSubmit={(evt) => {
+        evt.preventDefault();
+        if (note?.text) {
+          onSubmit(note);
+        }
+      }}
+    >
+      <TextEditor
+        clear={clear}
```

Comment:
text align boilerplate removed.
Re. key - that leads to jerky visual behaviour of the component because it's unmounting instead of hiding. Which can be fixed in other ways, but they would take much longer